### PR TITLE
Make token module public

### PR DESCRIPTION
**Pull request by pjoe**

### CODE REVIEW

**Comment by pjoe** [comment]
Looks like updated clippy found some issues, will take a look later

**Comment by Kangz** [comment]
LGTM! I don't know how comes tests were able to use the Token without this.

**Review by Kangz — APPROVED** [review]

File: src/lexer.rs [hunk 1]
```diff
@@ -308,11 +310,11 @@ impl<'a> Lexer<'a> {
                         _ => false,
                     })?
                 }
-                Some(('0'..='7', _)) => self.parse_number_radix(8, |c| c >= '0' && c <= '7')?,
+                Some(('0'..='7', _)) => self.parse_number_radix(8, |c| ('0'..='7').contains(&c))?,
```
Comment (Kangz) [inline src/lexer.rs:313]: Wow that's quite nice!

**Review by Kangz — APPROVED** [review]
Thanks for fixing clippy issues too!

**Pull request merged**